src/modules/Builder/hooks/useActivityFlowsRedirection.ts

```diff
@@ -14,16 +14,17 @@ export const useActivityFlowsRedirection = () => {
   const isNewApplet = useCheckIfNewApplet();
   const { watch } = useFormContext();
 
-  const activities = watch('activities');
-  const activityFlows = watch('activityFlows');
-  const activityFlow = activityFlows?.find(
-    (flow: ActivityFlowFormValues) => activityFlowId === getEntityKey(flow),
-  );
-  const shouldRedirect = (isNewApplet || activities?.length > 0) && activityFlowId && !activityFlow;
-
   useEffect(() => {
+    const activities = watch('activities');
```

Comment:
Is it ok watch inside useEffect?

Reply:
there are no errors, it seems it's just a function

Reply:
if use getValues instead of create subscriptions

Reply:
please don't unsubscribe then 
  // Callback version of watch.  It's your responsibility to unsubscribe when done.
  React.useEffect(() => {
    const subscription = watch((value, { name, type }) => console.log(value, name, type));
    return () => subscription.unsubscribe();
  }, [watch]);